pymc4/distributions/abstract/discrete.py

```diff
@@ -220,18 +206,16 @@ class Categorical(BoundedDiscreteDistribution):
     p : array of floats
         p > 0 and the elements of p must sum to 1. They will be automatically
         rescaled otherwise.
-
-    Developer Notes
-    ---------------
-    Parameter mappings to TensorFlow Probability are as follows:
-
-    - p: probs
     """
 
     def __init__(self, name, p, **kwargs):
         super().__init__(name, p=p, **kwargs)
 
-    # TODO: upper limit
+    def lower_limit(self):
+        return 0
+
+    def upper_limit(self):
+        return len(self.conditions["p"]) - 1
```

Comment:
this might behave inconsistently with broadcasting. We can make this method to be backend specific if generic `self.conditions["p"].shape[-1]` fails for some backend
```suggestion
        return self.conditions["p"].shape[-1] - 1
```